pkg/printers/github.go

```diff
@@ -1,28 +1,102 @@
+
+// GitHubProblemMatchers defines the root of problem matchers.
+// - https://github.com/actions/toolkit/blob/main/docs/problem-matchers.md
+// - https://github.com/actions/toolkit/blob/main/docs/commands.md#problem-matchers
+type GitHubProblemMatchers struct {
+	Matchers []GitHubMatcher `json:"problemMatcher,omitempty"`
+}
+
+// GitHubMatcher defines a problem matcher.
+type GitHubMatcher struct {
+	// Owner an ID field that can be used to remove or replace the problem matcher.
+	// **required**
+	Owner string `json:"owner,omitempty"`
+	// Severity indicates the default severity, either 'warning' or 'error' case-insensitive.
+	// Defaults to 'error'.
+	Severity string          `json:"severity,omitempty"`
+	Pattern  []GitHubPattern `json:"pattern,omitempty"`
+}
+
+// GitHubPattern defines a pattern for a problem matcher.
+type GitHubPattern struct {
+	// Regexp the regexp pattern that provides the groups to match against.
+	// **required**
+	Regexp string `json:"regexp,omitempty"`
+	// File a group number containing the file name.
+	File int `json:"file,omitempty"`
+	// FromPath a group number containing a filepath used to root the file (e.g. a project file).
+	FromPath int `json:"fromPath,omitempty"`
+	// Line a group number containing the line number.
+	Line int `json:"line,omitempty"`
+	// Column a group number containing the column information.
+	Column int `json:"column,omitempty"`
+	// Severity a group number containing either 'warning' or 'error' case-insensitive.
+	// Defaults to `error`.
+	Severity int `json:"severity,omitempty"`
+	// Code a group number containing the error code.
+	Code int `json:"code,omitempty"`
+	// Message a group number containing the error message.
+	// **required** at least one pattern must set the message.
+	Message int `json:"message,omitempty"`
+	// Loop whether to loop until a match is not found,
+	// only valid on the last pattern of a multi-pattern matcher.
+	Loop bool `json:"loop,omitempty"`
+}
+
 type GitHub struct {
 	w io.Writer
 }
 
-const defaultGithubSeverity = "error"
-
-// NewGitHub output format outputs issues according to GitHub actions format:
-// https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-error-message
+// NewGitHub output format outputs issues according to GitHub actions the problem matcher regexp.
 func NewGitHub(w io.Writer) *GitHub {
 	return &GitHub{w: w}
 }
 
-// print each line as: ::error file=app.js,line=10,col=15::Something went wrong
-func formatIssueAsGithub(issue *result.Issue) string {
-	severity := defaultGithubSeverity
+func (p *GitHub) Print(issues []result.Issue) error {
+	// Note: the file with the problem matcher definition should not be removed.
+	// A sleep can mitigate this problem but this will be flaky.
+	//
+	// Error: Unable to process command '::add-matcher::/tmp/golangci-lint-action-problem-matchers.json' successfully.
+	// Error: Could not find file '/tmp/golangci-lint-action-problem-matchers.json'.
+	//
```

Comment:
```suggestion
	// Result if the file is removed prematurely:
	// Error: Unable to process command '::add-matcher::/tmp/golangci-lint-action-problem-matchers.json' successfully.
	// Error: Could not find file '/tmp/golangci-lint-action-problem-matchers.json'.
```

Reply:
Those lines are real logs from GitHub Action, we cannot rephrase them.

Reply:
:thinking: the github diff is weird